source/extensions/filters/http/cache/cache_filter_logging_info.h

```diff
@@ -0,0 +1,115 @@
+#pragma once
+
+#include "envoy/stream_info/filter_state.h"
+
+#include "absl/strings/str_format.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace Cache {
+
+enum class LookupStatus {
+  // The CacheFilter couldn't determine the status of the request, probably
+  // because of an internal error.
+  Unknown,
+  // The CacheFilter found a response in cache to serve.
+  CacheHit,
+  // The CacheFilter didn't find a response in cache.
+  CacheMiss,
+  // The CacheFilter found a stale response, and sent a validation request to
+  // the upstream; the upstream responded with a 304 Not Modified.
+  StaleHitWithSuccessfulValidation,
+  // The CacheFilter found a stale response, and sent a validation request to
+  // the upstream; the upstream responded with anything other than a 304 Not
```

Comment:
i'm confused by the semantics here. If the upstream validates the request, why does the client care whether the cache hit required revalidation or not?

If the upstream (I assume that means origin from a cache perspective) send 5xx, wouldn't that be a validation failure?

Reply:
The client doesn't, but we do for metrics gathering purposes.